envoy/router/router.h

```diff
@@ -1335,6 +1334,8 @@ class GenericUpstream {
    * @param the account to assign the generic upstream.
    */
   virtual void setAccount(Buffer::BufferMemoryAccountSharedPtr account) PURE;
+
+  virtual void getStreamInfomation(StreamInfo::StreamInfo& stream_info) PURE;
```

Comment:
Maybe consider a name like `populateStreamInformation` or similar? getStreamInformation makes it seem like something is to be returned.